sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -400,6 +409,21 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     // metadata.
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
+  // Add optimization flags
+  // Assigning space for 16 characters.
+  char *backend_option = nullptr;
+  // Empty string is returned in backend_option when no appropriate backend
+  // option is available for a given opt level.
+  Plugin.getBackendOptimizationOption(optLevel, &backend_option);
+  if (backend_option && backend_option[0] != '\0') {
```

Comment:
maybe @steffenlarsen knows but do we have to use char* here? it would be a lot more simpler and cleaner with std::string

Reply:
I just followed the methodology used to support piPluginGetLastError. In this methodology, there is no memory allocation done in  SYCL runtime.

Thanks

Reply:
ah ok